Remove offensive language (Pt. 1)

Holzhaus opened the pull request:
This is a partial fix for #11931. What is missing is the CO group rename of `[Master]` and the `EngineMaster` rename, which would affect way more files.

Holzhaus committed:
refactor(enginemaster): Rename `MicMonitorMode::MASTER` to `Main`

Holzhaus committed:
refactor(enginechannel): Rename `master` CO to `main_mix`

Holzhaus committed:
refactor(enginemaster): Rename `getMasterBuffer` to `getMainBuffer`

Holzhaus committed:
refactor(engine): Replace term `MasterGain` with `MainGain`

Holzhaus committed:
refactor(enginemaster): Replace `pMaster` member names with `pMain`

Holzhaus committed:
chore(control): Replace term "master control" with "primary control"

Holzhaus committed:
refactor(enginemaster): Replace most instances of term `master` with `main`

Holzhaus committed:
chore(metaknob_link_test): Rename `m_master` to `m_main`

Holzhaus committed:
refactor(shoutconnection): Rename `masterSamplerate` to `mainSamplerate`

Holzhaus committed:
refactor(soundio): Rename `masterAudioLatency*` to `mainAudioLatency*`

Holzhaus committed:
chore(qml): Use new `sync_leader` CO for `SyncButton`

Holzhaus committed:
chore(schema): Fix mention of `main` branch in description

Holzhaus committed:
refactor(dlgprefsound): Replace term `master` with `main`

Swiftb0y commented on src/engine/enginemaster.cpp:
```diff
@@ -67,11 +67,11 @@ EngineMaster::EngineMaster(
     m_pWorkerScheduler->start(QThread::HighPriority);
 
     // Master sample rate
```
```suggestion
```

Swiftb0y commented on src/engine/enginemaster.cpp:
```diff
@@ -93,20 +93,20 @@ EngineMaster::EngineMaster(
     m_pBalance = new ControlPotmeter(ConfigKey(group, "balance"), -1., 1.);
 
     // Master gain
```
```suggestion
```

Swiftb0y reviewed (changes requested):
`AudioPath::AudioPathType` enum members (in `soundmanagerutil.h`) is not yet converted. Didn't do a thorough exhaustiveness check apart from that.

Holzhaus committed:
chore(enginemaster): Remove offensive language from comments

Holzhaus committed:
refactor(enginemaster): Rename `applyMasterEffects` to `applyMainEffects`

Holzhaus committed:
refactor(soundmanagerutil): Rename `AudioPathType::MASTER` enum member

Also, convert the legacy `enum` to a scoped `enum class` and switch the
enum to PascalCase.

Holzhaus commented:
> `AudioPath::AudioPathType` enum members (in `soundmanagerutil.h`) is not yet converted. Didn't do a thorough exhaustiveness check apart from that.

I added this in https://github.com/mixxxdj/mixxx/pull/11942/commits/43535a7c41c32363919b2ba41a8cb7a35930cd12. I think the PR is big enum now, let's do further changes in subsequent PRs to keep stuff reviewable.

Holzhaus commented:
Ah, and please don't merge this before 2.4 has been merged into main (if necessary). This will make it easier to resolve conflicts, because I guess there will be a bunch due to the channelcount/samplerate refactoring.

Swiftb0y commented:
> please merge this 2.4 into main before merging this PR (if necessary)

I assume this is not necessary anymore since you just did that an hour ago, right?

Swiftb0y commented on src/soundio/soundmanagerutil.h:
```diff
@@ -56,46 +56,46 @@ class AudioPath {
-    static unsigned char maxChannelsForType(AudioPathType type);
-
-    uint hashValue() const {
+  enum class AudioPathType : int {
```
any particular reason for specifying the underlying type explicitly?

Holzhaus replied:
> any particular reason for specifying the underlying type explicitly?

There is code that uses the enum's `int` value and converts between the types. Although the underlying type of a scoped `enum class` always `int` if not otherwise specified, I though that explicit is better than implicit and shows that we actually want to use `int` here.

Swiftb0y replied:
I think its only important to specify the underlying type when doing a bit_cast/reinterpret_cast on a struct that contains this. Otherwise the type is only specified to be as many bits wide as required to hold all enum values... 

Just because we static_cast from and to `int` for this enum doesn't really mean that it needs to be represented in memory as an `int`... Anyways, I don't really see the usage warranted here, but idc. Just my two cents.

Swiftb0y reviewed (approved):
LGTM. any objections?

Holzhaus committed:
chore(sounddevice): Fix outdated commented out variable names

Holzhaus commented:
I have a follow-up PR in progress where we rename to `EngineMixer` and a few other variable names and comments are fixed.

Swiftb0y commented:
Welp, here goes nothing. Lets see how many conflicts this creates.
